test/LevelZeroAcceleratorTopoTest.cpp

```diff
@@ -0,0 +1,241 @@
+    const int num_cpu = 56;
+    LevelZeroDevicePoolImp m_device_pool(*m_levelzero);
+
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR)).WillOnce(Return(num_accelerator));
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP)).WillOnce(Return(num_accelerator_subdevice));
+
+    LevelZeroAcceleratorTopo topo(m_device_pool, num_cpu);
+
+    EXPECT_EQ(num_accelerator, topo.num_accelerator());
+    EXPECT_EQ(num_accelerator_subdevice, topo.num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP));
+    std::set<int> cpus_allowed_set[num_accelerator];
+    cpus_allowed_set[0] = {0 ,1 ,2 ,3 ,4 ,5 ,6 };
+    cpus_allowed_set[1] = {7 ,8 ,9 ,10,11,12,13};
+    cpus_allowed_set[2] = {14,15,16,17,18,19,20};
+    cpus_allowed_set[3] = {21,22,23,24,25,26,27};
+    cpus_allowed_set[4] = {28,29,30,31,32,33,34};
+    cpus_allowed_set[5] = {35,36,37,38,39,40,41};
+    cpus_allowed_set[6] = {42,43,44,45,46,47,48};
+    cpus_allowed_set[7] = {49,50,51,52,53,54,55};
+
+    for (int accel_idx = 0; accel_idx < num_accelerator; ++accel_idx) {
+        ASSERT_THAT(topo.cpu_affinity_ideal(accel_idx), cpus_allowed_set[accel_idx]);
+        ASSERT_THAT(topo.cpu_affinity_ideal(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, accel_idx), cpus_allowed_set[accel_idx]);
+    }
+}
+
+//Test case: CPU count that is not evenly divisible by the accelerator count
+TEST_F(LevelZeroAcceleratorTopoTest, uneven_affinitization_config)
+{
+    const int num_accelerator = 3;
+    const int num_accelerator_subdevice = 6;
+    const int num_cpu =20;
+    LevelZeroDevicePoolImp m_device_pool(*m_levelzero);
+
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR)).WillOnce(Return(num_accelerator));
+    EXPECT_CALL(*m_levelzero, num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP)).WillOnce(Return(num_accelerator_subdevice));
+
+    LevelZeroAcceleratorTopo topo(m_device_pool, num_cpu);
+
+    EXPECT_EQ(num_accelerator, topo.num_accelerator());
+    std::set<int> cpus_allowed_set[num_accelerator];
+    cpus_allowed_set[0] = {0 ,1 ,2 ,3 ,4 ,5 ,18};
+    cpus_allowed_set[1] = {6 ,7 ,8 ,9 ,10,11,19};
+    cpus_allowed_set[2] = {12,13,14,15,16,17};
+
+    std::set<int> cpus_allowed_set_subdevice[num_accelerator_subdevice];
+    cpus_allowed_set_subdevice[0] = {0, 1, 2, 18};
+    cpus_allowed_set_subdevice[1] = {3, 4, 5, 19};
+    cpus_allowed_set_subdevice[2] = {6, 7, 8};
+    cpus_allowed_set_subdevice[3] = {9,10,11};
+    cpus_allowed_set_subdevice[4] = {12,13,14};
+    cpus_allowed_set_subdevice[5] = {14,16,17};
+
+    for (int accel_idx = 0; accel_idx < num_accelerator; ++accel_idx) {
+        ASSERT_THAT(topo.cpu_affinity_ideal(accel_idx), cpus_allowed_set[accel_idx]);
+    }
+    for (int accel_idx = 0; accel_idx < num_accelerator; ++accel_idx) {
+        ASSERT_THAT(topo.cpu_affinity_ideal(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP, accel_idx), cpus_allowed_set_subdevice[accel_idx]);
+    }
+
+    // TODO: Add subdevice check
```

Comment:
Issue for new test?

Reply:
Updated line 204 to cover sub-devices as intended.